test/integration/http2_integration_test.cc

```diff
@@ -70,6 +70,8 @@ TEST_P(Http2IntegrationTest, RetryAttemptCount) { testRetryAttemptCountHeader();
 
 TEST_P(Http2IntegrationTest, LargeRequestTrailersRejected) { testLargeRequestTrailers(66, 60); }
 
+TEST_P(Http2IntegrationTest, BasicMaxStreamDuration) { testMaxStreamDuration(); }
```

Comment:
If this is only implemented here, I think you want this inlined. However, I think you want this test for both HTTP/1 and HTTP/2 upstreams so I would put it in protocol_integration_test probably?